composeml/label_maker.py

```diff
@@ -254,98 +352,65 @@ def search(self,
         """Searches the data to calculates labels.
 
         Args:
-            df (DataFrame) : Data frame to search and extract labels.
-            num_examples_per_instance (int) : Number of examples per unique instance of target entity.
-            minimum_data (str) : Minimum data before starting search. Default value is first time of index.
-            gap (str or int) : Time between examples. Default value is window size.
+            df (DataFrame): Data frame to search and extract labels.
+            num_examples_per_instance (int): Number of examples per unique instance of target entity.
+            minimum_data (str): Minimum data before starting search. Default value is first time of index.
+            gap (str or int): Time between examples. Default value is window size.
                 If an integer, search will start on the first event after the minimum data.
-            drop_empty (bool) : Whether to drop empty slices. Default value is True.
-            label_type (str) : The label type can be "continuous" or "categorical". Default value is the inferred label type.
-            verbose (bool) : Whether to render progress bar. Default value is True.
-            *args : Positional arguments for labeling function.
-            **kwargs : Keyword arguments for labeling function.
+            drop_empty (bool): Whether to drop empty slices. Default value is True.
+            label_type (str): The label type can be "continuous" or "categorical". Default value is the inferred label type.
+            verbose (bool): Whether to render progress bar. Default value is True.
+            *args: Positional arguments for labeling function.
+            **kwargs: Keyword arguments for labeling function.
 
         Returns:
-            LabelTimes : Calculated labels with cutoff times.
+            lt (LabelTimes): Calculated labels with cutoff times.
         """
-        bar_format = "Elapsed: {elapsed} | Remaining: {remaining} | "
-        bar_format += "Progress: {l_bar}{bar}| "
-        bar_format += self.target_entity + ": {n}/{total} "
-        total = len(df.groupby(self.target_entity))
-        finite_examples_per_instance = num_examples_per_instance > -1 and num_examples_per_instance != float('inf')
+        assert self.labeling_function, 'missing labeling function(s)'
 
-        if finite_examples_per_instance:
-            total *= num_examples_per_instance
+        if self.window_size is None and gap is None:
```

Comment:
this check happens in a few places now. do we need multiple checks?

Reply:
Yes, this check should only happen in `search` and `slice` methods, but I can refactor into a reusable function.